pytorch/training/docker/1.6.0/py3/Dockerfile.gpu

```diff
@@ -0,0 +1,186 @@
+    build-essential \
+    ca-certificates \
+    cmake \
+    cuda-command-line-tools-10-1 \
+    cuda-cufft-10-1 \
+    cuda-curand-10-1 \
+    cuda-cusolver-10-1 \
+    cuda-cusparse-10-1 \
+    curl \
+    emacs \
+    git \
+    jq \
+    libglib2.0-0 \
+    libgl1-mesa-glx \
+    libsm6 \
+    libxext6 \
+    libxrender-dev \
+    libgomp1 \
+    libibverbs-dev \
+    libhwloc-dev \
+    libnuma1 \
+    libnuma-dev \
+    vim \
+    wget \
+    zlib1g-dev \
+ && apt-get remove -y cuda-cufft-dev-10-1 \
+    cuda-curand-dev-10-1 \
+    cuda-cusolver-dev-10-1 \
+    cuda-npp-dev-10-1 \
+    cuda-nvgraph-dev-10-1 \
+    cuda-nvjpeg-dev-10-1 \
+    cuda-nvrtc-dev-10-1
+
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/libcublas10_${CUBLAS_VERSION}.deb \
+ && dpkg -i libcublas10_${CUBLAS_VERSION}.deb \
+ && apt-get install -f -y \
+ && rm libcublas10_${CUBLAS_VERSION}.deb
+
+RUN wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && gunzip -c openmpi-$OPEN_MPI_VERSION.tar.gz | tar xf - \
+ && cd openmpi-$OPEN_MPI_VERSION \
+ && ./configure --prefix=/home/.openmpi \
+ && make all install \
+ && cd .. \
+ && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && rm -rf openmpi-$OPEN_MPI_VERSION
+
+ENV PATH="$PATH:/home/.openmpi/bin"
+ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+
+RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value \
+ && curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p /opt/conda \
+ && rm ~/miniconda.sh \
+ && /opt/conda/bin/conda install -y -c anaconda \
+    python=$PYTHON_VERSION \
+    numpy==1.16.4 \
+    ipython==7.10.1 \
+    mkl==2019.4 \
+    mkl-include==2019.4 \
+    cython==0.29.12 \
+    typing==3.6.4 \
+    future==0.17.1 \
+    "pyopenssl>=17.5.0" \
+ && conda install -c dglteam -y dgl-cuda10.1==0.4.3 \
+ && /opt/conda/bin/conda clean -ya
+
+RUN conda install -c pytorch magma-cuda101==2.5.2 \
+ && conda install -c conda-forge \
+    opencv==4.0.1 \
+ && conda install -y scikit-learn==0.21.2 \
+    pandas==0.25.0 \
+    h5py==2.9.0 \
+    requests==2.22.0 \
+    libgcc \
+ && conda clean -ya
+
+RUN pip install psutil==5.6.7 \
+                Pillow==7.1.0
+
+WORKDIR /opt/pytorch
+
+# Copy workaround script for incorrect hostname
+COPY changehostname.c /
+COPY start_with_right_hostname.sh /usr/local/bin/start_with_right_hostname.sh
+
+WORKDIR /root
+
+RUN /opt/conda/bin/conda config --set ssl_verify False \
+ && pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s /opt/conda/bin/pip /usr/local/bin/pip3
+
+# Uninstall torch and torchvision before installing the custom versions from an S3 bucket
+RUN pip install \
+    --no-cache-dir smdebug==${SMDEBUG_VERSION} \
+    "sagemaker<2" \
+    sagemaker-experiments==0.1.7 \
```

Comment:
want to double check: should this value be unpinned? Looks like there are many new patch versions of this library

Reply:
its pinned only for historical reasons, go ahead and unpin it

Reply:
to prevent a potential breaking change from a major version:
```pythonn
sagemaker-experiments==0.x
```

Reply:
if you don't include this change in this PR ill submit a follow up with it.

Reply:
thanks for the clarification, @choidongyeon can the change be made here and @danabens PR can address the pinnings in prior versions?

Reply:
1.6. changes next update, not resolving this so that @danabens sees @arjkesh's comment.